src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -519,13 +521,11 @@ private async Task<IEnumerable<Diagnostic>> GetProjectDiagnosticsAsync(Project p
 
                     // Have to see if this fix is still applicable.  Jump to the foreground thread
                     // to make that check.
-                    var applicable = await Task.Factory.StartNew(() =>
-                        {
-                            this.AssertIsForeground();
-                            return fix.Action.IsApplicable(document.Project.Solution.Workspace);
-                        },
-                        cancellationToken, TaskCreationOptions.None, ForegroundTaskScheduler).ConfigureAwait(false);
-                    this.AssertIsBackground();
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
+
+                    var applicable = fix.Action.IsApplicable(document.Project.Solution.Workspace);
+
+                    await TaskScheduler.Default;
```

Comment:
is there any concern, in general, about this await not happening (say, because of a cancellation exception).  Would we then be on some inappropriate thread?  Or does the 'await' of the caller ensure that doesn't happen?